### PR TITLE
Implement `require-button-type` rule

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/ember-template-lint/ember-template-lint/issues/867

### CODE REVIEW

File: test/unit/rules/lint-require-button-type-test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'require-button-type',
+
+  config: true,
+
+  good: [
```
Comment (bmish) [inline test/unit/rules/lint-require-button-type-test.js]: Should add valid test cases to ensure this doesn't affect other elements:

```
<div></div>
<div type="randomType"></div>
```

File: lib/rules/lint-require-button-type.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        let { chars } = value;
+        if (!['button', 'submit', 'reset'].includes(chars)) {
+          return this.logNode({ node, message: ERROR_MESSAGE });
+        }
+      },
+    };
+  }
+};
+
+module.exports.ERROR_MESSAGE = ERROR_MESSAGE;
```
Comment (bmish) [inline lib/rules/lint-require-button-type.js]: You're exporting the error message but you aren't actually importing it from the test file. Should update the test cases to use this.

File: test/unit/rules/lint-require-button-type-test.js (hunk 1 above)
Comment (Turbo87) [reply test/unit/rules/lint-require-button-type-test.js]: good idea, thanks :)

File: lib/rules/lint-require-button-type.js (hunk 2 above)
Comment (Turbo87) [reply lib/rules/lint-require-button-type.js]: done ✅

File: lib/rules/lint-require-button-type.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (dcyriller) [inline lib/rules/lint-require-button-type.js]: What do you think of using a static class field to export ERROR_MESSAGE?

Comment (Turbo87) [reply lib/rules/lint-require-button-type.js]: I think that's essentially what the code is currently doing. I don't think static class fields are supported by all of our currently supported Node versions though.

**Pull request merged**